src/panels/lovelace/entity-rows/hui-weather-entity-row.ts

```diff
@@ -0,0 +1,261 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  css,
+  CSSResult,
+  property,
+  customElement,
+  PropertyValues,
+} from "lit-element";
+
+import "../components/hui-generic-entity-row";
+import "../components/hui-warning";
+
+import { LovelaceRow, WeatherRowConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+import { hasConfigOrEntityChanged } from "../common/has-changed";
+import { classMap } from "lit-html/directives/class-map";
+import { actionHandler } from "../common/directives/action-handler-directive";
+import { hasAction } from "../common/has-action";
+import { ifDefined } from "lit-html/directives/if-defined";
+import { DOMAINS_HIDE_MORE_INFO } from "../../../common/const";
+import { computeDomain } from "../../../common/entity/compute_domain";
+import { computeStateName } from "../../../common/entity/compute_state_name";
+import { handleAction } from "../common/handle-action";
+import { ActionHandlerEvent } from "../../../data/lovelace";
+import { weatherIcons, cardinalDirections } from "../../../data/weather";
+
+@customElement("hui-weather-entity-row")
+class HuiWeatherEntityRow extends LitElement implements LovelaceRow {
+  @property() public hass?: HomeAssistant;
+
+  @property() private _config?: WeatherRowConfig;
+
+  public setConfig(config: WeatherRowConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = {
+      primary_attribute: "extrema",
+      secondary_attribute: "humidity",
```

Comment:
I would rather this be precipitation but this attribute is only accessible via sensors

Reply:
It is in the forecast for some weather platforms, let's just use it. And let's not make these settings.
If precipitation is available use that, otherwise use humidity.